cmd/serverless/main.go

```diff
@@ -228,13 +221,13 @@ func runAgent(ctx context.Context, stopCh chan struct{}) (err error) {
 		}
 	}
 
-	// read configuration from the environment vars
-	// --------------------------------------------
+	// read configuration from both the environment vars and the config file
+	// if one is provided
+	// --------------------------
 
-	// note that this call is counter-intuitive: it must return an error because
-	// no files should exist, and then, the configuration is read from env vars.
-	if _, confErr := config.Load(); confErr == nil {
-		log.Warn("A configuration file has been found, which should not happen in this mode.")
+	config.Datadog.SetConfigFile(datadogConfigPath)
+	if _, confErr := config.LoadWithoutSecret(); confErr == nil {
```

Comment:
Could people add their secret executable to the lambda root and be able to use the secret function ? This might be for another PR.

Reply:
Discussed offline but for now, we won't use the `secrets` feature in the Lambda Extension. It should not be a big deal to enable it but it would be for another PR 👍